prog/vnet/nic_nexus.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::NicNexus < Prog::Base
+  semaphore :destroy, :refresh_mesh, :detach_vm
+
+  def self.assemble(private_subnet_id, name: nil, ipv6_addr: nil, ipv4_addr: nil)
+    unless (subnet = PrivateSubnet[private_subnet_id])
+      fail "Given subnet doesn't exist with the id #{private_subnet_id}"
+    end
+
+    ubid = UBID.generate(UBID::TYPE_NIC)
+    name ||= Nic.ubid_to_name(ubid)
+
+    Validation.validate_name(name)
+
+    ipv6_addr ||= subnet.random_private_ipv6.to_s
+    ipv4_addr ||= subnet.random_private_ipv4.to_s
+
+    DB.transaction do
+      nic = Nic.create(private_ipv6: ipv6_addr, private_ipv4: ipv4_addr, mac: gen_mac,
+        encryption_key: gen_encryption_key, name: name,
+        private_subnet_id: private_subnet_id) { _1.id = ubid.to_uuid }
+      subnet.add_nic(nic)
+      Strand.create(prog: "Vnet::NicNexus", label: "wait_vnet") { _1.id = ubid.to_uuid }
+    end
+  end
+
+  def self.gen_encryption_key
+    "0x" + SecureRandom.bytes(36).unpack1("H*")
+  end
+
+  def nic
+    @nic ||= Nic[strand.id]
+  end
+
+  def wait_vnet
+    unless nic.private_subnet.strand.label == "wait"
+      nap 1
```

Comment:
'nother one of these loops (not your fault, but it suggests we lack a programming model we need)